_books/ion-1-1/src/text/values.md

```diff
@@ -0,0 +1,489 @@
+# Values
+
+## Annotations
+In the text format, type annotations are denoted by a [symbol token](symbol-tokens.md) and double-colons preceding any value.
+Multiple annotations on the same value are separated by double-colons:
+
+```ion
+int32::12                                // Suggests 32 bits as end-user type
+degrees::'celsius'::100                  // You can have multiple annotaions on a value
+'my.custom.type'::{ x : 12 , y : -1 }    // Gives a struct a user-defined type
+
+{ field: some_annotation::value }        // Field's name must precede annotations of its value
+
+jpeg :: {{ ... }}                        // Indicates the blob contains jpeg data
+bool :: null.int                         // A very misleading annotation on the integer null
+'' :: 1                                  // An empty annotation
+null.symbol :: 1                         // ERROR: type annotation cannot be null 
+```
+
+## Nulls
+
+Null values are represented by the keyword `null`, optionally followed by `.` and the name of a type in the Ion data model.
+
+```ion
+null
+null.null       // Identical to unadorned null
+null.bool
+null.int
+null.float
+null.decimal
+null.timestamp
+null.string
+null.symbol
+null.blob
+null.clob
+null.struct
+null.list
+null.sexp
+```
+
+The text format treats all of these as reserved tokens; to use those
+same characters as a symbol token, they must be enclosed in single-quotes:
+
+```ion
+null        // The type is null
+'null'      // The type is symbol
+null.list   // The type is list
+'null.int'  // The type is symbol
+```
+
+Any text token starting with `null.` must be one of the legal null values.
+
+```ion
+(llun.foo)  // A s-expression equivalent to (llun . foo)
+
+(null.foo)  // This is illegal; not equivalent to (null . foo) 
+            // because null. is never split into separate tokens
+```
+
+## Booleans
+
+Boolean values are represented by the literals `true` and `false`.
+
+The text format treats both of these as reserved tokens; to use those
+same characters as a symbol token, they must be enclosed in single-quotes.
+
+```ion
+true          // a boolean value
+'true'        // a symbol value
+
+'true'::1     // an integer annotated with the text "true"
+true::1       // ERROR: cannot use an unquoted keyword as an annotation
+
+{ 'true': 1 } // a struct containing a field name with the text "true"
+{ true: 1 }   // ERROR: cannot use an unquoted keyword as a field name
+```
+
+## Integers
+
+Integer values may be encoded in binary, decimal, and hexadecimal notation.
```

Comment:
Should we add some note in this section that integers provided as tagless arguments have limited range?

Reply:
That's covered in the chapter on Macros.